File: javascript/house/house.js
const verses = [
  null,
  { thing: 'house that Jack built', action: null },
  { thing: 'malt', action: 'lay in' },
  { thing: 'rat', action: 'ate' },
  { thing: 'cat', action: 'killed' },
  { thing: 'dog', action: 'worried' },
  { thing: 'cow with the crumpled horn', action: 'tossed' },
  { thing: 'maiden all forlorn', action: 'milked' },
  { thing: 'man all tattered and torn', action: 'kissed' },
  { thing: 'priest all shaven and shorn', action: 'married' },
  { thing: 'rooster that crowed in the morn', action: 'woke' },
  { thing: 'farmer sowing his corn', action: 'kept' },
  { thing: 'horse and the hound and the horn', action: 'belonged to' }
];

const House = {
  verse: (num) => {
    const res = [`This is the ${verses[num].thing}${num == 1 ? '.' : ''}`];
    while (num > 1) {
      res.push(
        `that ${verses[num].action} the ${verses[num - 1].thing}${num === 2 ? '.' : ''}`
      );
      num -= 1;
    }
    return res;
  },
  verses: (start, end) => {
    let res = []
    for (let i = start; i <= end; i++) {
      res = [...res, ...House.verse(i), ''];
    }
    return res.slice(0, -1);
  },
};

export default House;

File: java/house/src/main/java/House.java
import java.util.stream.IntStream;
import java.util.stream.Collectors;

class House {
  private final String[][] verses = { { null },
    { "house that Jack built.", null },
    { "malt ", "lay in" },
    { "rat ", "ate" },
    { "cat ", "killed" },
    { "dog ", "worried" },
    { "cow with the crumpled horn ", "tossed" },
    { "maiden all forlorn ", "milked" },
    { "man all tattered and torn ", "kissed" },
    { "priest all shaven and shorn ", "married" },
    { "rooster that crowed in the morn ", "woke" },
    { "farmer sowing his corn ", "kept" },
    { "horse and the hound and the horn ", "belonged to" },
  };

  String verse(int n) {
    StringBuilder res = new StringBuilder("This is the " + verses[n][0]);
    for (int i = n; i > 1; i--) {
      res.append("that " + verses[i][1] + " the " + verses[i - 1][0]);
    }
    return res.toString();
  }

  String verses(int start, int end) {
    return IntStream.rangeClosed(start, end)
      .mapToObj(i -> verse(i))
      .collect(Collectors.joining("\n"));
  }

  String sing() {
    return verses(1, 12);
  }
}